code/modules/plumbing/sink.dm

```diff
@@ -0,0 +1,181 @@
+/obj/structure/sink
+	name = "sink"
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "sink"
+	desc = "A sink used for washing one's hands and face."
+	anchored = 1
+	var/busy = 0 	//Something's being washed at the moment
+	var/use_reagents = TRUE//If we're reliant on the plumbing system
+
+/obj/structure/sink/Initialize()
+	create_reagents(15)
+	..()
+
+/obj/structure/sink/attack_hand(mob/living/user)
+	if(!user || !istype(user))
+		return
+	if(!iscarbon(user))
+		return
+	if(!Adjacent(user))
+		return
+	if(!reagents.total_volume && !plumbing_has_reagents(15) && use_reagents) //15 units per use
+		to_chat(user, "<span class='warning'>Nothing comes out of the spigot!</span>")
+		return
+	if(busy)
+		to_chat(user, "<span class='notice'>Someone's already washing here.</span>")
+		return
+	if(use_reagents)
+		master_plumber.request_liquid(src, 15)
+	var/selected_area = parse_zone(user.zone_selected)
+	var/washing_face = 0
+	if(selected_area in list("head", "mouth", "eyes"))
+		washing_face = 1
+	user.visible_message("<span class='notice'>[user] starts washing their [washing_face ? "face" : "hands"]...</span>", \
+						"<span class='notice'>You start washing your [washing_face ? "face" : "hands"]...</span>")
+	busy = 1
```

Comment:
This is legacy code that was moved to a new file. It'll be touched up after I migrate the other objects.